### ISSUE TITLE
Gherkin cache invalid since v4.5

### ISSUE THREAD

**Issue by mihaeu** [Behat/Gherkin#128]
Hey everyone, thanks for this awesome project!

We noticed that our Behat tests on Jenkins were slowly dying and after a couple of hours of debugging we noticed that the problem has something to do with outdated caches since the 4.5 update.

What happened was that you could run the whole unit as one, but when you tried to launch a single .feature file Behat was unable to find any specifications.

I deleted the caches and downgraded to Gherkin v4.4 and everything is back to normal. Did we miss something or was there a breaking change?

Cheers,
Mike

**Comment by pfrenssen** [comment]
I'm having the same issue. I cannot run individual tests, getting this error:

```
 [Behat\Testwork\Tester\Exception\WrongPathsException]                                                                                           
  No specifications found at path(s) `features/authentication.feature`. This might be because of incorrect paths configuration in your `suites`.
```

**Comment by pfrenssen** [comment]
Note for people who may not be able to run their tests even after rolling back to 4.4.5: delete the Gherkin cache, this is located in your temporary storage, e.g. in `/tmp/behat_gherkin_cache/` or similar.

**Comment by pfrenssen** [comment]
I've been experimenting with this. The bug is apparently not in `behat/gherkin` but in `behat/behat`. The bug only occurs if I am using Behat 3.4.0, regardless of the version of Gherkin.

**Comment by pfrenssen** [comment]
I did a `git bisect` and the bug was introduced in this commit from `behat/behat`: https://github.com/Behat/Behat/commit/08802b7f2d736b88df339e5411a1c722e92adbe3

This was part of https://github.com/Behat/Behat/pull/1056. We should move this discussion to `behat/behat`.

**Comment by pfrenssen** [comment]
It appears that the bug is in Gherkin after all. I proposed a fix in #129 but this needs some input because I am looking at the Gherkin code base for the first time and don't know its intended architecture.

**Comment by kwoxer** [comment]
Thanks. Indeed this is a version Issue. For me it worked to have this version setup active:
    "behat/behat": "3.3.0",
    "behat/gherkin": "4.4.5",
    "drupal/drupal-extension": "3.3.0",
    "behat/symfony2-extension": "2.1.1"

**Comment by phil-davis** [comment]
Is this fixed by #166 release 4.6.2 ?
